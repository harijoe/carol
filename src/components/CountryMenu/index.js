import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { getUrlFromLocale } from 'utils/locale'

const CountryOption = ({ name, value }) =>
  <FormattedMessage id={name} key={value}>
    {message =>
      <option value={value}>
        {message}
      </option>}
  </FormattedMessage>

CountryOption.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
}

const CountryMenu = ({ currentLocale, className }) =>
  <select
    defaultValue={currentLocale}
    className={className}
    onChange={e => {
      location.href = (locale => getUrlFromLocale(locale))(e.target.value)
    }}
  >
    <CountryOption name="country.spain" value="es-ES" />
    <CountryOption name="country.france" value="fr-FR" />
    <CountryOption name="country.great_britain" value="en-GB" />
  </select>

CountryMenu.propTypes = {
  currentLocale: PropTypes.string,
  className: PropTypes.string,
}

export default CountryMenu
