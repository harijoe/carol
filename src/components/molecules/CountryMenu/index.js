import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { getUrlFromLocale } from 'utils/locale'
import { port, ssl } from 'config'

const homePageUrl = (locale) => {
  const standardPortUsed = (ssl.enabled && port === 443) || (!ssl.enabled && port === 80)

  return getUrlFromLocale(locale)
    .replace(/https?/, ssl.enabled ? 'https' : 'http')
    .replace(/(:\d+)?$/, standardPortUsed ? '' : `:${port}`)
}

const CountryOption = ({ name, value }) =>
  <FormattedMessage id={name} key={value}>
    {message => <option value={value}>{message}</option>}
  </FormattedMessage>

CountryOption.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
}

const CountryMenu = ({ currentLocale, className }) =>
  <select defaultValue={currentLocale} className={className} onChange={(e) => { location.href = homePageUrl(e.target.value) }}>
    <CountryOption name="country.spain" value="es_ES" />
    <CountryOption name="country.france" value="fr_FR" />
    <CountryOption name="country.great_britain" value="en_GB" />
  </select>

CountryMenu.propTypes = {
  currentLocale: PropTypes.string,
  className: PropTypes.string,
}

export default CountryMenu
