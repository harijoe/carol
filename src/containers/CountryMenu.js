import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import CountryMenu from 'components/CountryMenu'
import { fromContext } from 'store/selectors'

const CountryMenuContainer = props => <CountryMenu {...props} />

CountryMenuContainer.propTypes = {
  locale: PropTypes.string,
}

const mapStateToProps = state => ({ currentLocale: fromContext.getLocale(state) })

export default connect(mapStateToProps)(CountryMenuContainer)
