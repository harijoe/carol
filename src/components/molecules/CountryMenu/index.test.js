import React from 'react'
import { shallow } from 'enzyme'

import CountryMenu from '.'
import intlMock from '../../../../test/intlMock'

const wrap = (props = {}) => shallow(intlMock(<CountryMenu {...props} />))

it('renders CountryMenu', () => {
  expect(wrap()).toMatchSnapshot()
})
