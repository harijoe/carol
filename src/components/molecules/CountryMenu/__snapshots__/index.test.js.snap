exports[`test renders CountryMenu 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CountryMenu />,
  "nodes": Array [
    <CountryMenu />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <IntlProvider
        locale="en"
        messages={
                Object {
                        "Invalid username and password combination": "Invalid username and password combination",
                        "Missing parameters. \"username\" and \"password\" required": "Missing parameters. \"username\" and \"password\" required",
                        "burger_menu": "Burger Menu",
                        "construction": "Construction",
                        "country.france": "France",
                        "country.great_britain": "Great-Britain",
                        "country.spain": "Spain",
                        "firm.details.budget": "Budget",
                        "firm.details.certificate": "Certificate",
                        "firm.details.city": "City",
                        "firm.details.city_code": "City code",
                        "firm.details.client_since": "Member since",
                        "firm.details.conviviality": "Conviviality",
                        "firm.details.country": "Country",
                        "firm.details.delay": "Delay",
                        "firm.details.description": "Description",
                        "firm.details.employees_number": "Number of employee",
                        "firm.details.home_improvement_id": "Home Improvement Id",
                        "firm.details.logo": "Logo",
                        "firm.details.logo_missing": "No logo",
                        "firm.details.name": "Firm Name",
                        "firm.details.notation": "Notation",
                        "firm.details.postal_code": "Postal code",
                        "firm.details.registration_number": "Registration number",
                        "firm.details.served_area_cities": "Served Area Postal Codes",
                        "firm.details.website": "Website",
                        "firm.details.website_missing": "No website",
                        "firm.find_pro": "Find a pro",
                        "firm.i_am_pro": "I am a pro",
                        "firm.resource": "Resource for my work",
                        "firm.site_search": "Site search",
                        "firm.site_search.result_link": "More details",
                        "fixing": "Fixing",
                        "form.input.button.phone": "Phone",
                        "help": "Help",
                        "home": "Home",
                        "login": "Login",
                        "project.created_at": "Created at",
                        "project.elaboration.response": "Response",
                        "project.elaboration.submit": "Submit",
                        "project.elaboration.title": "Project Elaboration",
                        "project.list.empty": "No project yet",
                        "project.list.heading": "My projects",
                        "project.reference": "Reference",
                        "project.status": "Status",
                        "project.status.completion_in_progress": "Projet en cours d’élaboration EN",
                        "project.status.finished": "Projet terminé EN",
                        "project.status.found": "Projet en cours de recherche de pros : X PRO(S) TROUVÉS EN",
                        "project.status.pending_search": "Projet en cours de recherche de pros : pas de pro trouvé EN",
                        "project.status.started": "Projet démarré entre un artisan et un particulier EN",
                        "project.status.to_validate": "Projet élaboré mais user non auto-validé EN",
                        "project.submit_button": "OK EN",
                        "project.submit_failed": "Projet n\'a pas été envoyé! EN",
                        "project.submit_failed_title": "Oups! EN",
                        "project.submit_success": "Projet envoyé! EN",
                        "project.submit_success_title": "Félicitations! EN",
                        "project.updated_at": "Updated at",
                        "project.validate_link": "Validate",
                        "project.validation_page": "Page Validation EN",
                        "renovation": "Renovation",
                        "search": "Search",
                        "server_error": "An error occured, please try again later",
                        "test": "test",
                        "user.address": "Address",
                        "user.birthday": "Birthday",
                        "user.captcha_error": "Are you a bot?",
                        "user.choose_img": "Choose a picture",
                        "user.city": "City",
                        "user.confirm_password": "Confirm your password",
                        "user.contact_info": "Contact infos",
                        "user.continue_with_facebook": "Continue with Facebook",
                        "user.continue_with_google": "Continue with Google",
                        "user.country_code": "Country",
                        "user.create_account": "I want to create an account",
                        "user.drop_img": "Try dropping some files here, or click to select files to upload.",
                        "user.email": "Email",
                        "user.facebook_id": "Facebook id",
                        "user.favorite_page": "Favorite page",
                        "user.first_name": "First Name",
                        "user.fixed_phone": "Phone",
                        "user.forget_password": "Forget password ?",
                        "user.forgot_password.heading": "Forgotten password",
                        "user.gender": "Gender",
                        "user.google_id": "Google id",
                        "user.invalid_login": "Your login and/or password are not valid",
                        "user.last_name": "Last Name",
                        "user.log_out": "Log out",
                        "user.message_page": "My Message page",
                        "user.messaging": "Messaging",
                        "user.mobile_phone": "Mobile phone number",
                        "user.mr": "Mr",
                        "user.mrs": "Mrs",
                        "user.my_account": "My account",
                        "user.my_projects": "My projects",
                        "user.newsletter_subscription": "I want to subscribe to Quotatis weekly newsletter",
                        "user.password": "Password",
                        "user.password_match_err": "Passwords do not match",
                        "user.postal_code": "Postal code",
                        "user.prefered_language": "Prefered language",
                        "user.profile_info": "Profile infos",
                        "user.project_page": "My project page",
                        "user.region": "Region",
                        "user.reset_password_email": "An email was sent to reset your password.",
                        "user.reset_password_success": "Your password has been successfully changed.",
                        "user.send": "Send",
                        "user.sign_in": "Sign in",
                        "user.sign_up": "Sign up",
                        "user.sign_up.confirmation": "An email was sent to confirm your account.",
                        "user.sign_up.heading": "Subscription page",
                        "user.social_network": "Social Networks Parameters",
                        "user.thank_you": "Thank you!",
                        "user.update": "Update",
                        "validators.forgot_password.unexpired_token": "You already asked to reinitialize your password, please check your emails.",
                        "validators.forgot_password.user_not_found": "User not found",
                        "validators.integer": "Must be an integer",
                        "validators.max_length": "Must be no more than {max} characters",
                        "validators.min_length": "Must be at least {min} characters",
                        "validators.must_be_one_of": "Must be one of {values}",
                        "validators.must_match": "Must match",
                        "validators.required_field": "Required field",
                        "validators.user.already_exist": "This user already exists",
                        "validators.user.firstName.not_blank": "Field \"First name\" has not to be blank",
                        "validators.user.fixedPhone.invalid_type": "Field \"Fixe phone\" is not valid",
                        "validators.user.gender.invalid_choice": "Field \"Gender\" is not valid",
                        "validators.user.gender.not_blank": "Field \"Gender\" has not to be blank",
                        "validators.user.invalid_email": "Invalid email address",
                        "validators.user.invalid_url": "Invalid url",
                        "validators.user.lastName.not_blank": "Field \"Last name\" has not to be blank",
                        "validators.user.mobilePhone.invalid_type": "Field \"Mobile phone\" is not valid",
                        "validators.user.mobilePhone.not_blank": "Field \"Mobile phone\" has not to be blank",
                        "validators.user.password.not_blank": "Field \"Password\" has not to be blank",
                        "validators.user.postalCode.not_blank": "Field \"Postal code\" has not to be blank",
                      }
        }>
        <CountryMenu />
</IntlProvider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": IntlProvider {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "intl": undefined,
        },
        "props": Object {
          "children": <CountryMenu />,
          "locale": "en",
          "messages": Object {
            "Invalid username and password combination": "Invalid username and password combination",
            "Missing parameters. \"username\" and \"password\" required": "Missing parameters. \"username\" and \"password\" required",
            "burger_menu": "Burger Menu",
            "construction": "Construction",
            "country.france": "France",
            "country.great_britain": "Great-Britain",
            "country.spain": "Spain",
            "firm.details.budget": "Budget",
            "firm.details.certificate": "Certificate",
            "firm.details.city": "City",
            "firm.details.city_code": "City code",
            "firm.details.client_since": "Member since",
            "firm.details.conviviality": "Conviviality",
            "firm.details.country": "Country",
            "firm.details.delay": "Delay",
            "firm.details.description": "Description",
            "firm.details.employees_number": "Number of employee",
            "firm.details.home_improvement_id": "Home Improvement Id",
            "firm.details.logo": "Logo",
            "firm.details.logo_missing": "No logo",
            "firm.details.name": "Firm Name",
            "firm.details.notation": "Notation",
            "firm.details.postal_code": "Postal code",
            "firm.details.registration_number": "Registration number",
            "firm.details.served_area_cities": "Served Area Postal Codes",
            "firm.details.website": "Website",
            "firm.details.website_missing": "No website",
            "firm.find_pro": "Find a pro",
            "firm.i_am_pro": "I am a pro",
            "firm.resource": "Resource for my work",
            "firm.site_search": "Site search",
            "firm.site_search.result_link": "More details",
            "fixing": "Fixing",
            "form.input.button.phone": "Phone",
            "help": "Help",
            "home": "Home",
            "login": "Login",
            "project.created_at": "Created at",
            "project.elaboration.response": "Response",
            "project.elaboration.submit": "Submit",
            "project.elaboration.title": "Project Elaboration",
            "project.list.empty": "No project yet",
            "project.list.heading": "My projects",
            "project.reference": "Reference",
            "project.status": "Status",
            "project.status.completion_in_progress": "Projet en cours d’élaboration EN",
            "project.status.finished": "Projet terminé EN",
            "project.status.found": "Projet en cours de recherche de pros : X PRO(S) TROUVÉS EN",
            "project.status.pending_search": "Projet en cours de recherche de pros : pas de pro trouvé EN",
            "project.status.started": "Projet démarré entre un artisan et un particulier EN",
            "project.status.to_validate": "Projet élaboré mais user non auto-validé EN",
            "project.submit_button": "OK EN",
            "project.submit_failed": "Projet n\'a pas été envoyé! EN",
            "project.submit_failed_title": "Oups! EN",
            "project.submit_success": "Projet envoyé! EN",
            "project.submit_success_title": "Félicitations! EN",
            "project.updated_at": "Updated at",
            "project.validate_link": "Validate",
            "project.validation_page": "Page Validation EN",
            "renovation": "Renovation",
            "search": "Search",
            "server_error": "An error occured, please try again later",
            "test": "test",
            "user.address": "Address",
            "user.birthday": "Birthday",
            "user.captcha_error": "Are you a bot?",
            "user.choose_img": "Choose a picture",
            "user.city": "City",
            "user.confirm_password": "Confirm your password",
            "user.contact_info": "Contact infos",
            "user.continue_with_facebook": "Continue with Facebook",
            "user.continue_with_google": "Continue with Google",
            "user.country_code": "Country",
            "user.create_account": "I want to create an account",
            "user.drop_img": "Try dropping some files here, or click to select files to upload.",
            "user.email": "Email",
            "user.facebook_id": "Facebook id",
            "user.favorite_page": "Favorite page",
            "user.first_name": "First Name",
            "user.fixed_phone": "Phone",
            "user.forget_password": "Forget password ?",
            "user.forgot_password.heading": "Forgotten password",
            "user.gender": "Gender",
            "user.google_id": "Google id",
            "user.invalid_login": "Your login and/or password are not valid",
            "user.last_name": "Last Name",
            "user.log_out": "Log out",
            "user.message_page": "My Message page",
            "user.messaging": "Messaging",
            "user.mobile_phone": "Mobile phone number",
            "user.mr": "Mr",
            "user.mrs": "Mrs",
            "user.my_account": "My account",
            "user.my_projects": "My projects",
            "user.newsletter_subscription": "I want to subscribe to Quotatis weekly newsletter",
            "user.password": "Password",
            "user.password_match_err": "Passwords do not match",
            "user.postal_code": "Postal code",
            "user.prefered_language": "Prefered language",
            "user.profile_info": "Profile infos",
            "user.project_page": "My project page",
            "user.region": "Region",
            "user.reset_password_email": "An email was sent to reset your password.",
            "user.reset_password_success": "Your password has been successfully changed.",
            "user.send": "Send",
            "user.sign_in": "Sign in",
            "user.sign_up": "Sign up",
            "user.sign_up.confirmation": "An email was sent to confirm your account.",
            "user.sign_up.heading": "Subscription page",
            "user.social_network": "Social Networks Parameters",
            "user.thank_you": "Thank you!",
            "user.update": "Update",
            "validators.forgot_password.unexpired_token": "You already asked to reinitialize your password, please check your emails.",
            "validators.forgot_password.user_not_found": "User not found",
            "validators.integer": "Must be an integer",
            "validators.max_length": "Must be no more than {max} characters",
            "validators.min_length": "Must be at least {min} characters",
            "validators.must_be_one_of": "Must be one of {values}",
            "validators.must_match": "Must match",
            "validators.required_field": "Required field",
            "validators.user.already_exist": "This user already exists",
            "validators.user.firstName.not_blank": "Field \"First name\" has not to be blank",
            "validators.user.fixedPhone.invalid_type": "Field \"Fixe phone\" is not valid",
            "validators.user.gender.invalid_choice": "Field \"Gender\" is not valid",
            "validators.user.gender.not_blank": "Field \"Gender\" has not to be blank",
            "validators.user.invalid_email": "Invalid email address",
            "validators.user.invalid_url": "Invalid url",
            "validators.user.lastName.not_blank": "Field \"Last name\" has not to be blank",
            "validators.user.mobilePhone.invalid_type": "Field \"Mobile phone\" is not valid",
            "validators.user.mobilePhone.not_blank": "Field \"Mobile phone\" has not to be blank",
            "validators.user.password.not_blank": "Field \"Password\" has not to be blank",
            "validators.user.postalCode.not_blank": "Field \"Postal code\" has not to be blank",
          },
        },
        "refs": Object {},
        "state": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
          "now": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CountryMenu />,
        "_debugID": 2,
        "_renderedOutput": <CountryMenu />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <IntlProvider
    locale="en"
    messages={
        Object {
            "Invalid username and password combination": "Invalid username and password combination",
            "Missing parameters. \"username\" and \"password\" required": "Missing parameters. \"username\" and \"password\" required",
            "burger_menu": "Burger Menu",
            "construction": "Construction",
            "country.france": "France",
            "country.great_britain": "Great-Britain",
            "country.spain": "Spain",
            "firm.details.budget": "Budget",
            "firm.details.certificate": "Certificate",
            "firm.details.city": "City",
            "firm.details.city_code": "City code",
            "firm.details.client_since": "Member since",
            "firm.details.conviviality": "Conviviality",
            "firm.details.country": "Country",
            "firm.details.delay": "Delay",
            "firm.details.description": "Description",
            "firm.details.employees_number": "Number of employee",
            "firm.details.home_improvement_id": "Home Improvement Id",
            "firm.details.logo": "Logo",
            "firm.details.logo_missing": "No logo",
            "firm.details.name": "Firm Name",
            "firm.details.notation": "Notation",
            "firm.details.postal_code": "Postal code",
            "firm.details.registration_number": "Registration number",
            "firm.details.served_area_cities": "Served Area Postal Codes",
            "firm.details.website": "Website",
            "firm.details.website_missing": "No website",
            "firm.find_pro": "Find a pro",
            "firm.i_am_pro": "I am a pro",
            "firm.resource": "Resource for my work",
            "firm.site_search": "Site search",
            "firm.site_search.result_link": "More details",
            "fixing": "Fixing",
            "form.input.button.phone": "Phone",
            "help": "Help",
            "home": "Home",
            "login": "Login",
            "project.created_at": "Created at",
            "project.elaboration.response": "Response",
            "project.elaboration.submit": "Submit",
            "project.elaboration.title": "Project Elaboration",
            "project.list.empty": "No project yet",
            "project.list.heading": "My projects",
            "project.reference": "Reference",
            "project.status": "Status",
            "project.status.completion_in_progress": "Projet en cours d’élaboration EN",
            "project.status.finished": "Projet terminé EN",
            "project.status.found": "Projet en cours de recherche de pros : X PRO(S) TROUVÉS EN",
            "project.status.pending_search": "Projet en cours de recherche de pros : pas de pro trouvé EN",
            "project.status.started": "Projet démarré entre un artisan et un particulier EN",
            "project.status.to_validate": "Projet élaboré mais user non auto-validé EN",
            "project.submit_button": "OK EN",
            "project.submit_failed": "Projet n\'a pas été envoyé! EN",
            "project.submit_failed_title": "Oups! EN",
            "project.submit_success": "Projet envoyé! EN",
            "project.submit_success_title": "Félicitations! EN",
            "project.updated_at": "Updated at",
            "project.validate_link": "Validate",
            "project.validation_page": "Page Validation EN",
            "renovation": "Renovation",
            "search": "Search",
            "server_error": "An error occured, please try again later",
            "test": "test",
            "user.address": "Address",
            "user.birthday": "Birthday",
            "user.captcha_error": "Are you a bot?",
            "user.choose_img": "Choose a picture",
            "user.city": "City",
            "user.confirm_password": "Confirm your password",
            "user.contact_info": "Contact infos",
            "user.continue_with_facebook": "Continue with Facebook",
            "user.continue_with_google": "Continue with Google",
            "user.country_code": "Country",
            "user.create_account": "I want to create an account",
            "user.drop_img": "Try dropping some files here, or click to select files to upload.",
            "user.email": "Email",
            "user.facebook_id": "Facebook id",
            "user.favorite_page": "Favorite page",
            "user.first_name": "First Name",
            "user.fixed_phone": "Phone",
            "user.forget_password": "Forget password ?",
            "user.forgot_password.heading": "Forgotten password",
            "user.gender": "Gender",
            "user.google_id": "Google id",
            "user.invalid_login": "Your login and/or password are not valid",
            "user.last_name": "Last Name",
            "user.log_out": "Log out",
            "user.message_page": "My Message page",
            "user.messaging": "Messaging",
            "user.mobile_phone": "Mobile phone number",
            "user.mr": "Mr",
            "user.mrs": "Mrs",
            "user.my_account": "My account",
            "user.my_projects": "My projects",
            "user.newsletter_subscription": "I want to subscribe to Quotatis weekly newsletter",
            "user.password": "Password",
            "user.password_match_err": "Passwords do not match",
            "user.postal_code": "Postal code",
            "user.prefered_language": "Prefered language",
            "user.profile_info": "Profile infos",
            "user.project_page": "My project page",
            "user.region": "Region",
            "user.reset_password_email": "An email was sent to reset your password.",
            "user.reset_password_success": "Your password has been successfully changed.",
            "user.send": "Send",
            "user.sign_in": "Sign in",
            "user.sign_up": "Sign up",
            "user.sign_up.confirmation": "An email was sent to confirm your account.",
            "user.sign_up.heading": "Subscription page",
            "user.social_network": "Social Networks Parameters",
            "user.thank_you": "Thank you!",
            "user.update": "Update",
            "validators.forgot_password.unexpired_token": "You already asked to reinitialize your password, please check your emails.",
            "validators.forgot_password.user_not_found": "User not found",
            "validators.integer": "Must be an integer",
            "validators.max_length": "Must be no more than {max} characters",
            "validators.min_length": "Must be at least {min} characters",
            "validators.must_be_one_of": "Must be one of {values}",
            "validators.must_match": "Must match",
            "validators.required_field": "Required field",
            "validators.user.already_exist": "This user already exists",
            "validators.user.firstName.not_blank": "Field \"First name\" has not to be blank",
            "validators.user.fixedPhone.invalid_type": "Field \"Fixe phone\" is not valid",
            "validators.user.gender.invalid_choice": "Field \"Gender\" is not valid",
            "validators.user.gender.not_blank": "Field \"Gender\" has not to be blank",
            "validators.user.invalid_email": "Invalid email address",
            "validators.user.invalid_url": "Invalid url",
            "validators.user.lastName.not_blank": "Field \"Last name\" has not to be blank",
            "validators.user.mobilePhone.invalid_type": "Field \"Mobile phone\" is not valid",
            "validators.user.mobilePhone.not_blank": "Field \"Mobile phone\" has not to be blank",
            "validators.user.password.not_blank": "Field \"Password\" has not to be blank",
            "validators.user.postalCode.not_blank": "Field \"Postal code\" has not to be blank",
          }
    }>
    <CountryMenu />
</IntlProvider>,
}
`;
